File: tests/fixtures/wait_for_debugger.py
#!/usr/bin/env python3
"""
Test program that waits for a debugger to attach.
Runs a debugpy server in listen mode.
"""
import sys
import debugpy

# Listen for debugger connection
port = int(sys.argv[1]) if len(sys.argv) > 1 else 5678
print(f"Debugpy listening on port {port}", flush=True)

try:
    debugpy.listen(("127.0.0.1", port))
except Exception as e:
    print(f"Failed to listen on port {port}: {e}", flush=True)
    sys.exit(1)

print("Waiting for debugger to attach...", flush=True)
debugpy.wait_for_client()
print("Debugger attached!", flush=True)

# Simple code to debug
x = 42
y = x + 1  # Breakpoint here
print(f"Result: {y}")

File: tests/fixtures/wait_for_debugger.js
// Test program that waits for a debugger to attach
// Run with: node --inspect-brk=port wait_for_debugger.js

console.log("Node.js process started, waiting for debugger...");

// Give debugger time to attach
setTimeout(() => {
  console.log("Debugger should be attached now");
  
  // Simple code to debug
  const x = 42;
  const y = x + 1;  // Breakpoint here (line 11)
  console.log(`Result: ${y}`);
}, 100);
